litellm/llms/bedrock_httpx.py

```diff
@@ -247,10 +252,57 @@ def get_credentials(
             aws_session_name,
             aws_profile_name,
             aws_role_name,
+            aws_web_identity_token,
         ) = params_to_check
 
         ### CHECK STS ###
-        if aws_role_name is not None and aws_session_name is not None:
+        if aws_web_identity_token is not None and aws_role_name is not None and aws_session_name is not None:
+            iam_creds_cache_key = json.dumps({
+                "aws_web_identity_token": aws_web_identity_token,
+                "aws_role_name": aws_role_name,
+                "aws_session_name": aws_session_name,
+                "aws_region_name": aws_region_name,
+            })
+
+            iam_creds_dict = iam_cache.get_cache(iam_creds_cache_key)
+            if iam_creds_dict is None:
+                oidc_token = get_secret(aws_web_identity_token)
```

Comment:
fixing this - https://github.com/BerriAI/litellm/issues/3858, is a precursor to this PR, since this would also introduce sync flows to an async call